csi/volumeutils.py

```diff
@@ -1009,13 +1015,224 @@ def reload_glusterfs(volume):
+        with open(volfile) as volf:
+            for line in volf:
+                line = line.strip()
+                if line.startswith("volume "):
+                    element = VolfileElement(line.split()[-1])
+                    #print(line.split()[-1])
+                elif line == "end-volume":
+                    elements.append(element)
+                    element = None
+                elif line.startswith("option "):
+                    _, name, value = line.split()
+                    element.options[name] = value
+                elif line.startswith("subvolumes "):
+                    element.subvolumes = line.split(" ", 1)[-1]
+                elif line.startswith("type "):
+                    element.type = line.split()[-1]
+                    # print("type")
+                    # print(line.split()[-1])
+        return Volfile(volfile, elements)
+
+    def update_options_by_type1(self, opts):
+        for element in self.elements:
+            if opts.get(element.type, None) is not None:
+                logging.info(logf(
+                    "opts[element.type]",
+                    opts_element_type=opts[element.type],
+                    element_type=element.type
+                ))
+                element.options.update(opts[element.type])
+                logging.info(logf(
+                    "opts[element.type]",
+                    opts_element_type=opts[element.type],
+                    element_type=element.type
+                ))
+
+    def update_options_by_type2(self, opts):
+        opt = {}
+        for element in self.elements:
+
+            # Example: 'cluster/replicate' & 'performance/open-behind'
+            # Below logic to get only 'replicate' from 'cluster/replicate',
+            # Exception being for 'performance/open-behind' keep only 'performance',
+            # As vol-options under type 'performance' starts with it.
+            e_type = [element.type.split("/")[0] if "performance" in element.type.split("/")\
+                    else element.type.split("/")[-1]][0]
+            for types, values in opts.items():
+                #print(element.type, element.options, types, values)
+                if e_type in types:
+
+                    # #without check for individual option
+                    # print("before update", element.name, element.type)
+                    # print(element.options)
+                    # print("\n")
+                    # #option = {}
+                    # #option[key] = value
+                    # #print("values, values[key]", values, values[key], option)
+                    # element.options.update(values)
+                    # print("after update", element.name, element.type)
+                    # print(element.options)
+                    # print("\n")
+                    # opt[element.type] = element.options
+                    # # #values = {}, ele.options = {}
+
+                    for key, value in values.items():
```

Comment:
This check was required to update the option in the element if it only exists, else would add to all matched elements.